deepset_cloud_sdk/_api/files.py

```diff
@@ -117,3 +123,80 @@ async def list_paginated(
         data = response_body["data"]
         has_more = response_body["has_more"]
         return FileList(total=total, data=[File.from_dict(d) for d in data], has_more=has_more)
+
+    @staticmethod
+    def _available_file_name(file_path: Path, suffix: str = "_1") -> str:
+        logger.warning("File already exists. Renaming file to avoid overwriting.", file_path=str(file_path))
+        base, ext = os.path.splitext(str(file_path))
+        new_filename = f"{base}{suffix}{ext}"
+        while os.path.exists(new_filename):
+            suffix = f"_{int(suffix[1:]) + 1}"
+            new_filename = f"{base}{suffix}{ext}"
```

Comment:
optional: I think this would be all a bit nicer / more readable by using `pathlib`

Reply:
same for all the code below

Reply:
done :)